wespeaker/dataset/dataset_utils.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2024 Hongji Wang (jijijiang77@gmail.com)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import torch
+
+
+def apply_cmvn(feats, norm_mean=True, norm_var=False):
+    # feats batch: (B,T,F)
+    if norm_mean:
+        feats = feats - torch.mean(feats, dim=1, keepdim=True)
+    if norm_var:
+        feats = feats / torch.sqrt(torch.var(feats, dim=1, keepdim=True) + 1e-7)
+
+    return feats
+
+
+def spec_aug(feats, num_t_mask=1, num_f_mask=1, max_t=10, max_f=8, prob=0.6):
+    # feats batch: (B,T,F)
+    # do spec_aug on all batch samples using a same group of params randomly
+    # TODO (hongji): do spec_aug on each sample separately
```

Comment:
I think you can directly use the implementation in https://pytorch.org/audio/master/generated/torchaudio.transforms.FrequencyMasking.html#torchaudio.transforms.FrequencyMasking

Reply:
Good idea. I will try it later.